### ISSUE TITLE
Unions containg never can now be constructed via DAA, which violates assumption that never expressions are unreachable

### ISSUE THREAD

**Issue by DenialAdams** [DenialAdams/roland#123]
```roland
union U {
   a: bool,
   b: !,
}
proc main() {
   let x: Union;
   x.a = 10;
   x.b;
   // Roland assumes anything hereafter is unreachable, but it will be reached
}
```

**Comment by DenialAdams** [comment]
We have this same problem with:

```
proc main() {
   let a = 10;
   let x = gen$<!>(&a);
}

proc gen<T>(x: &u8) -> &T {
   return x transmute &T;
}
```

I think we need to simply legislate this: an expression producing a type of never will trap

**Comment by DenialAdams** [comment]
Closed in 8bb875a817abe4817ab5837b74df53171a18c390
